torch/distributed/fsdp/flat_param.py

```diff
@@ -1517,12 +1530,24 @@ def _use_sharded_flat_param(self) -> None:
             )
         flat_param.data = flat_param._local_shard  # type: ignore[attr-defined]
         if self._use_orig_params:
-            self._use_sharded_views()
+            in_forward = self._training_state == HandleTrainingState.FORWARD
```

Comment:
does this work for activation checkpointing as well? or would we need AC to support a recomputing flag?

Reply:
I think it should work with AC for the cases we have been using FSDP <> AC. Previously, we already directly check `self._training_state == HandleTrainingState.FORWARD` to tell if we are in forward, regardless of recomputing or not. I will see if the unit tests fail though.